### PR TITLE
cleanup fields

### LINKED ISSUE #21: User login with session, user API, new routes
**Is your feature request related to a problem? If so, please describe.**
I started working on a login/user profile feature for the front-end. This requires an API
**Describe the solution you'd like**
A user API with the following routes:
- `/api/user/login` and `/api/user/register`
- `/api/user/modules` to list all modules
- `/api/user/:username` to get userinfo

An `email` field would also be great, because let's face it, yes we want privacy, but for account recoveries, etc. we need the users' email.

Obivously, this is not hight-priority, don't want to ruin your roadmap, but we'll need to implement this in the future.

### PR DESCRIPTION
**Pull request by divy-work**
Closes #21 

cc @zorbyte

### CODE REVIEW

**Comment by divy-work** [comment]
oops, i always forget we have badges for `feat` lol

File: src/main.rs [hunk 1]
```diff
@@ -150,14 +150,22 @@ async fn upload_package(
...
     .await;
     match package_uploaded {
-      Ok(_) => Ok(
-        HttpResponse::Ok()
-          .content_type("application/json")
-          .body(serde_json::to_string(&UploadResult { ok: true })?),
-      ),
+      Ok(_) => Ok(HttpResponse::Ok().content_type("application/json").body(
+        serde_json::to_string(&UploadResult {
+          ok: true,
```
Comment (zorbyte) [inline src/main.rs:155]: 1. To keep this consistent with x's RESTful responses, could the following format be implemented:
```json
{
  "code": 200,
  "msg": "String stuff"
}
```

**Comment by divy-work** [comment]
@zorbyte I've also added a `Config` struct that would be used in future to implement remote configs.

File: src/main.rs [hunk 2]
```diff
@@ -150,14 +150,22 @@ async fn upload_package(
...
+        })?,
+      )),
       Err(e) => {
         println!("{:?}", &e);
-        Ok(HttpResponse::InternalServerError().into())
+        Ok(
+          HttpResponse::InternalServerError()
+            .content_type("application/json")
+            .body(serde_json::to_string(&UploadResult {
+              ok: false,
```
Comment (zorbyte) [inline src/main.rs:165]: 1.

File: src/main.rs [hunk 3]
```diff
@@ -69,13 +69,13 @@ pub struct PackageConfig {
 #[derive(serde::Serialize, Debug)]
 pub struct UploadResult {
   ok: bool,
```
Comment (zorbyte) [inline src/main.rs:71]: 1.

File: src/main.rs (hunk 1 above)
Comment (divy-work) [reply src/main.rs:155]: Sure!

**Comment by divy-work** [comment]
@zorbyte Updated.

**Pull request merged**